pyzo/pyzokernel/interpreter.py

```diff
@@ -437,13 +437,20 @@ def _run_startup_code(self, startup_info):
         
         if '\n' in script:
             # Run code later or now
-            firstline = script.split('\n')[0].replace(' ', '')
-            if firstline.startswith('#AFTER_GUI'):
-                self._codeToRunOnStartup = script
+            scriptSplit = script.split('\n')
+            foundAfter = False
+            for idx, line in enumerate(scriptSplit):
+                if line.replace(' ', '').startswith('#AFTER_GUI'):
+                    foundAfter = True
+                    break
+            if foundAfter:
+                self._codeToRunOnStartup = '\n'.join(scriptSplit[idx::])
+                runBefore = '\n'.join(scriptSplit[0:idx])
             else:
-                self.context._stat_interpreter.send('Busy')
-                msg = {'source': script, 'fname': '<startup>', 'lineno': 0}
-                self.runlargecode(msg, True)
+                runBefore = script
+            self.context._stat_interpreter.send('Busy')
+            msg = {'source': runBefore, 'fname': '<startup>', 'lineno': 0}
+            self.runlargecode(msg, True)
```

Comment:
Good start! I'd like these last three lines to run only when there is actually any code to run *before*. Something like this, for instance:
```py
linesBefore = []
linesAfter = script.splitlines()
while linesAfter:
    if linesAfter[0].startswith('#AFTER_GUI'):
        break
    linesBefore.append(linesAfter.pop(0))
scriptBefore ='\n'.join(linesBefore)
scriptAfter='\n'.join(linesAfter)

if scriptBefore.strip():  # don't trigger when only empty lines
    ...
 ```

Reply:
Sorry for almost completely rewriting your contribution. 

It would be sufficient to only put the three lines above in a ``if scriptBefore.strip(): ``